src/cli.ts

```diff
@@ -67,7 +67,8 @@ export async function run(): Promise<void> {
   const serverInfos = await startServers(options);
 
   if (serverInfos.kind === 'mainline') {
-    const urls = getServerUrls(options, serverInfos[0]!.server);
+    const mainlineServer = serverInfos;
```

Comment:
Ouch. I guess we get that from not having strict indexing types. You don't need to do this assignment, though it does help legibility.